src/deepsparse/server/openai_server.py

```diff
@@ -0,0 +1,385 @@
+        def tokenize(text: str) -> List[int]:
+            return pipeline.tokenizer(text)
+
+        prompt_token_ids = tokenize(prompt)
+        generation_kwargs = map_generation_schema(generation_kwargs)
+
+        stream = generation_kwargs["stream"]
+        presence_penalty = generation_kwargs["presence_penalty"]
+        stop = generation_kwargs["stop"]
+
+        generation_kwargs.pop("stream")
+        generation_kwargs.pop("presence_penalty")
+        generation_kwargs.pop("stop")
+
+        output = pipeline(
+            sequences=prompt,
+            generation_config=generation_kwargs,
+            streaming=stream,
+            presence_penalty=presence_penalty,
+            stop=stop,
+        )
+
+        if not stream:
+            # Non-streaming responss
+            generations = output.generations[0]
+            if not isinstance(generations, list):
+                generations = [generations]
+
+            generated_outputs = []
+            for prompt_generation in generations:
+                completion = CompletionOutput(
+                    index=0,
+                    text=prompt_generation.text,
+                    token_ids=tokenize(prompt_generation.text),
+                    finish_reason=prompt_generation.finished_reason,
+                )
+                generated_outputs.append(completion)
+            yield RequestOutput(
+                request_id=request_id,
+                prompt=prompt,
+                prompt_token_ids=prompt_token_ids,
+                outputs=generated_outputs,
+                finished=True,
+            )
+        else:
+            concat_text = ""
+            concat_token_ids = []
+            for generation in output:
+                output = generation.generations[0]
+                concat_text += output.text
+                concat_token_ids.append(tokenize(output.text))
+                yield RequestOutput(
+                    request_id=request_id,
+                    prompt=prompt,
+                    prompt_token_ids=prompt_token_ids,
+                    outputs=[
+                        CompletionOutput(
+                            index=0,
+                            text=concat_text,
+                            token_ids=concat_token_ids,
+                            finish_reason=output.finished_reason,
+                        )
+                    ],
+                    finished=output.finished,
+                )
+
+    @staticmethod
+    async def show_available_models(proxy_pipeline: ProxyPipeline):
+        """Show available models. Right now we only have one model."""
+        model_cards = [
+            ModelCard(
+                id=proxy_pipeline.pipeline.model_path,
+                root=proxy_pipeline.pipeline.model_path,
+                permission=[ModelPermission()],
+            )
+        ]
+        return ModelList(data=model_cards)
+
+    @staticmethod
+    async def create_chat_completion(
```

Comment:
should have a way to extend the pipeline timer so we could potentially benchmark the routines in this function if wanted

Reply:
anything that screams out at you as providing issues in adding this in? Happy to do this in a follow-up PR

Reply:
no can add in here. thinking more for the refactor